docs/source/reference/ha.rst

```diff
@@ -0,0 +1,301 @@
+:orphan:
+
+High availability deployment
+============================
+
+|st2| has been systematically built with High availability(HA) as a goal. The exact deployment
+steps to achieve HA depend on the specifics of the infrastructure in which |st2| is deployed. This
+guide covers a brief explanation on various |st2| services, how they interact and the external
+services necessary for |st2| to function.
+
+In the section :doc:`/install/overview` a detailed picture and explanation on how single-box deployments
+work is provided. Repeating the picture here to keep context and use as a reference to layer on some
+HA deployment specific details.
+
+.. figure :: /_static/images/st2-deployment-big-picture.png
+    :align: center
+
+.. source https://docs.google.com/drawings/d/1X6u8BB9bnWkW8C81ERBvjIKRfo9mDos4XEKeDv6YiF0/edit
+
+Components
+----------
+
+st2api
+^^^^^^
+This process host the REST API endpoints that serve requests from  WebUI, CLI and ChatOps. It maintains
+connections to MongoDB to store and retrieve information. It also connects to RabbitMQ to push various
+kind of messages onto the message bus. It is a gunicorn managed process which by default listens on
```

Comment:
Maybe we could mention it's a Python WSGI app running under gunicorn, but that's not a big deal.